app/routes.py

```diff
@@ -1,2 +1,312 @@
+            description=request_body["description"],
+            completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        new_task_response = {"task":{
+                "id": new_task.id,
+                "title" : new_task.title,
+                "description": new_task.description,
+                "is_complete": False if new_task.completed_at == None else True 
+                }
+        }
+        return jsonify(new_task_response), 201 
+    
+
+    if request.method == "GET":
+        
+        sort_query = request.args.get("sort") # create a variable called sort and assign request.args.get which gives me access to everything after the ? and everything after the question mark is a key value pair 
+        tasks = Task.query.all()
+        tasks_response = []
+        for task in tasks:
+                tasks_response.append(
+                    {
+
+                    "id": task.id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True 
+
+
+                    })
+        
+        if sort_query == "asc":
+            # tasks = Task.query.order_by(Task.title.asc())
+            tasks_response.sort(key=lambda x: x["title"])
+
+        if sort_query == "desc" :
+            tasks_response.sort(key=lambda x: x["title"], reverse=True)
+            # .query.order_by("title".desc())
+                
+        # if sort_title_query:
+        #     if sort_title_query == "asc":
+        #         tasks = Task.query.order_by(Task.title.asc())
+        #     elif sort_title_query == "desc":
+        #         tasks = Task.query.order_by(Task.title.desc())
+        return jsonify(tasks_response)
+    
+    
+
+    
+@tasks_bp.route("/<task_id>", methods = ["GET", "PUT", "DELETE"])
+def handle_task(task_id): # cannot have the same handle 
+
+    task = Task.query.get(task_id)# retrieve one task instance 
+    if task is None:
+        return jsonify(None), 404
+
+    elif request.method == "GET":
+        if not task.goal_id:
+        
+            return {"task":{
+                    "id": task.id,
+                    "title" : task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True 
+                    }
+            }, 200
+        else:
+            return {"task":{
+                    "id": task.id,
+                    "goal_id": task.goal.id,
+                    "title" : task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True 
+                    }
+            }, 200
+
+    
+    elif request.method == "PUT":
+        form_data = request.get_json() #this function takes the json data and turns it into a python dictionary
+        print(form_data)
```

Comment:
It is best practice to do a final run-through of your code (remove print statements, commented out code, remove unused imports, etc) before submitting it for review.